src/components/ProbeEditor/ProbeEditor.tsx

```diff
@@ -31,9 +32,11 @@ export const ProbeEditor = ({
   probe,
   submitText,
   supportingContent,
+  readOnly,
```

Comment:
Can we add a few tests as a safety and implicit documentation for the `readOnly` prop?

My worry is reading this in the future it is confusing  there is a `readOnly` prop and also a `useCanEditProbe` hook so it looks like something is redundant and can be removed.

I think much of the confusion stems from the component names. Should we make a `ViewProbe` component to add to the router? Maybe we can tackle that when we do https://github.com/grafana/synthetic-monitoring-app/issues/848?

Reply:
- updated the prop name `forceViewMode`
- added test for the new prop

This is an intermediate solution for the lack of a `ViewProbe` component.